File: source/Emphasize/Format Definitions/BoldItalic.vb
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Utilities
Imports System.ComponentModel.Composition


Namespace FormatDefinitions

    <ClassificationType(ClassificationTypeNames:=BoldItalic.Name)>
    <Export(GetType(EditorFormatDefinition))>
    <Name(BoldItalic.Name)>
    <Order(After:=Priority.High)>
    <UserVisible(True)>
    Public NotInheritable Class BoldItalic
        Inherits ClassificationFormatDefinition


        Public Const Name As String = "Comment - Bold, Italic Span"


        <Export(GetType(ClassificationTypeDefinition))>
        <BaseDefinition("Comment")>
        <Name(Name)>
        Public Shared ReadOnly ClassificationType As ClassificationTypeDefinition


        Public Sub New()
            DisplayName = Name
            IsBold = True
            IsItalic = True
        End Sub

    End Class

End Namespace

File: source/Emphasize/FormatDefinitions/BoldItalic.cs
#nullable enable

using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Utilities;
using System.ComponentModel.Composition;


namespace Emphasize.FormatDefinitions {

    [ClassificationType(ClassificationTypeNames = Name)]
    [Export(typeof(EditorFormatDefinition))]
    [Name(Name)]
    [Order(After = Priority.High)]
    [UserVisible(true)]
    public class BoldItalic : ClassificationFormatDefinition {

        public const string Name = "Emphasize - Bold, Italic";


        [Export(typeof(ClassificationTypeDefinition))]
        [Name(Name)]
        public static readonly ClassificationTypeDefinition? ClassificationType;


        public BoldItalic() {
            DisplayName = Name;
            IsBold = true;
            IsItalic = true;
        }

    }

}
